File: vlsisapd/examples/agds/python/driveAgds.py
import AGDS
lib = AGDS.Library("myTestLib")
lib.setUserUnits(0.001)
lib.setPhysUnits(1.0e-9)

active = AGDS.Rectangle( 6, 120, 290, 540, 690) # layer  6 corresponds to active
poly   = AGDS.Rectangle(17, 305, 150, 365, 830) # layer 17 corresponds to polysilicium

str = AGDS.Structure("Transistor")
str.addElement(active)
str.addElement(poly)

lib.addStructure(str)
lib.writeToFile("./transistor.agds")

File: deprecated/vlsisapd/examples/agds/cplusplus/driveAgds.cpp
#include <string>
using namespace std;

#include "vlsisapd/agds/Library.h"
#include "vlsisapd/agds/Structure.h"
#include "vlsisapd/agds/Rectangle.h"

int main(int argc, char * argv[]) {
    AGDS::Library* lib = new AGDS::Library(string("myTestLib"));

    lib->setUserUnits(0.001);
    lib->setPhysUnits(1.0E-9);

    AGDS::Rectangle* poly   = new AGDS::Rectangle( 17, 305, 150, 365, 830 );
    AGDS::Rectangle* active = new AGDS::Rectangle(  6, 130, 290, 540, 690 );

    AGDS::Structure* str = new AGDS::Structure("Transistor");

    str->addElement(poly);
    str->addElement(active);

    lib->addStructure(str);

    lib->writeToFile("./transistor.agds");
    
    return 0;
}

